src/interface/meshblock_data.cpp

```diff
@@ -16,45 +16,55 @@
 #include <cstdlib>
 #include <memory>
 #include <sstream>
+#include <unordered_set>
 #include <utility>
 #include <vector>
 
 #include "bvals/cc/bvals_cc.hpp"
+#include "interface/metadata.hpp"
+#include "interface/state_descriptor.hpp"
+#include "interface/variable.hpp"
+#include "interface/variable_pack.hpp"
 #include "mesh/mesh.hpp"
 #include "mesh/meshblock.hpp"
+#include "utils/error_checking.hpp"
 
 namespace parthenon {
 
 template <typename T>
-void MeshBlockData<T>::Add(const std::vector<std::string> &labelArray,
-                           const Metadata &metadata) {
-  // generate the vector and call Add
-  for (const auto &label : labelArray) {
-    Add(label, metadata);
+void MeshBlockData<T>::Initialize(
+    const std::shared_ptr<StateDescriptor> resolved_packages,
+    const std::shared_ptr<MeshBlock> pmb) {
+  SetBlockPointer(pmb);
+  resolved_packages_ = resolved_packages;
```

Comment:
You mean in the constructor? We can't set it in the constructor because `MeshBlockData` needs to be default-constructible for `DataCollection`.

I'm not very happy with this and would prefer if only valid `MeshBlockData` instances were created with a mesh block pointer and a resolved package, but I think that would require a lot of changes in the code base...

Reply:
Oh oops my bad. Somehow I thought this was a constructor. Ignore me. ;)